File: LivrusVirtus/FoxToolKit/FXRuby/ConIdiomas.rb
require 'fox'
require 'fox/responder'
require 'resource'
require 'Principal'
require 'ConsultasSQL'
require 'Consultas'
include Fox
include Responder

class FRMConIdiomas < FRMConsultas
  
  def initialize(owner)  
    super(owner, Resource::STR_CONIDIOMA)

    FXMAPFUNC(SEL_COMMAND, ID_FRMCONSULTAS_BTNPESQUISAR, :onBtnPesquisar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCONSULTAS_BTNLIMPAR, :onBtnLimpar)
    FXMAPFUNC(SEL_SELECTED, ID_FRMCONSULTAS_GRIDCONSULTA, :onGridSelected)
    FXMAPFUNC(SEL_CHANGED, ID_FRMCONSULTAS_GRIDCONSULTA, :onGridChanged)
    
    @labDescricao.setText("Idioma:")
    @gridConsulta.setItemText(0, 2, "Idioma")
  end

  def create()
    super
    show(PLACEMENT_SCREEN)
  end
  
  def onBtnPesquisar(sender, sel, ptr)
    PesquisarDados($ConsSQL.Idioma('S',"", 
      @edtDescricao.getText()), Resource::STR_IDINENC, 2)
    return true
  end
  
  def onBtnLimpar(sender, sel, ptr)
    LimparDados()
    return true
  end
  
  def onGridSelected(sender, sel, ptr)
    setLabRegistros(ptr.fm.row)
    return true
  end
  
  def onGridChanged(sender, sel, ptr)
    setLabRegistros(ptr.row)
    return true
  end

end

File: LivrusVirtus/FoxToolKit/FXPython/ConIdiomas.py
from FXPy.fox import *
from Consultas import frmConsultas
from resource import Resource
from ConsultasSQL import ConsultasSQL
import Principal

class frmConIdiomas(frmConsultas):

  def __init__(self, owner):
    frmConsultas.__init__(self, owner, Resource.STR_CONIDIOMA)  

    FXMAPFUNC(self, SEL_COMMAND, frmConIdiomas.ID_FRMCONSULTAS_BTNPESQUISAR, 
      frmConIdiomas.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmConIdiomas.ID_FRMCONSULTAS_BTNLIMPAR, 
      frmConIdiomas.onBtnLimpar)
    FXMAPFUNC(self, SEL_SELECTED, frmConIdiomas.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConIdiomas.onGridSelected)
    FXMAPFUNC(self, SEL_CHANGED, frmConIdiomas.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConIdiomas.onGridChanged)
    
    self.labDescricao.setText("Idioma:")
    self.gridConsulta.setItemText(0, 2, "Idioma")

  def create(self):
    frmConsultas.create(self)
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(ConsultasSQL.ConsSQL.Idioma('S',"", 
      self.edtDescricao.getText()), Resource.STR_IDINENC, 2)
    return True
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.LimparDados()
    return True
  
  def onGridSelected(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
  
  def onGridChanged(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
